src/openrct2/ride/TrackData.cpp

```diff
@@ -5428,6 +5428,745 @@ constexpr static uint8_t TrackTypeToSpinFunction[TrackElemType::Count] = {
     NO_SPIN, NO_SPIN, NO_SPIN, NO_SPIN, NO_SPIN, NO_SPIN, NO_SPIN, NO_SPIN, NO_SPIN
 };
 
+template<int32_t TConstant> static int32_t EvaluatorConst(const int16_t)
+{
+    return TConstant;
+}
+
+static int32_t EvaluatorVerticalLoop(const int16_t progress)
+{
+    return (abs(progress - 155) / 2) + 28;
+}
+
+static int32_t EvaluatorHalfLoopUp(const int16_t progress)
+{
+    return ((static_cast<uint16_t>(-(progress - 155))) / 2) + 28;
```

Comment:
What if the result is negative? The cast will return a crazy value.

Reply:
It's the exact same as how it was before this PR: https://github.com/OpenRCT2/OpenRCT2/pull/15401/files#diff-e80b0c445c9e2b2302d2ddd41d9feded6f6b18f9d68c38d49d02fc52b5061cd7L5863